src/test/java/org/opensearchmetrics/util/OpenSearchUtilTest.java

```diff
@@ -48,25 +56,47 @@ void setUp() {
     void WHEN_index_exists_THEN_doNothing() throws IOException {
         when(client.indices()).thenReturn(indicesClient);
         when(indicesClient.exists(any(GetIndexRequest.class), any())).thenReturn(true);
-        openSearchUtil.createIndexIfNotExists("some_index");
+        openSearchUtil.createIndexIfNotExists("some_index", Optional.empty());
 
         verify(indicesClient, times(0)).create(any(CreateIndexRequest.class), any(RequestOptions.class));
     }
 
     @Test
-    void WHEN_index_not_exist_THEN_create_index() throws IOException {
+    void WHEN_index_not_exist_THEN_create_index_along_with_alias() throws IOException {
 
         when(client.indices()).thenReturn(indicesClient);
         when(indicesClient.exists(any(GetIndexRequest.class), any(RequestOptions.class))).thenReturn(false);
         when(indicesClient.create(any(CreateIndexRequest.class), any(RequestOptions.class)))
                 .thenReturn(new CreateIndexResponse(true, true, "some_index"));
-        openSearchUtil.createIndexIfNotExists("some_index");
+        when(indicesClient.updateAliases(any(IndicesAliasesRequest.class), any(RequestOptions.class)))
+                .thenReturn(new AcknowledgedResponse(true));
+        openSearchUtil.createIndexIfNotExists("some_index", Optional.of("maintainer-activity"));
 
         verify(indicesClient).exists(any(GetIndexRequest.class), any(RequestOptions.class));
         verify(indicesClient).create(any(CreateIndexRequest.class), any(RequestOptions.class));
+        verify(indicesClient).updateAliases(any(IndicesAliasesRequest.class), any(RequestOptions.class));
         verifyNoMoreInteractions(indicesClient);
     }
 
+    @Test
+    void WHEN_index_not_exist_THEN_create_index_along_with_alias_exception() throws IOException {
+        when(client.indices()).thenReturn(indicesClient);
+        when(indicesClient.exists(any(GetIndexRequest.class), any(RequestOptions.class))).thenReturn(false);
+        when(indicesClient.create(any(CreateIndexRequest.class), any(RequestOptions.class)))
+                .thenReturn(new CreateIndexResponse(true, true, "some_index"));
+
+        doThrow(new IOException("Error adding alias to index")).when(indicesClient).updateAliases(any(IndicesAliasesRequest.class), any(RequestOptions.class));
+
+        try {
+            openSearchUtil.createIndexIfNotExists("some_index", Optional.of("maintainer-activity"));
+            fail("Expected a RuntimeException to be thrown");
+        } catch (RuntimeException e) {
+            // Exception caught as expected
+            System.out.println("Caught exception message: " + e.getMessage());
```

Comment:
Can we remove this `System.out.println` ?

Reply:
Sure Prudhvi